src/index.ts

```diff
@@ -1,27 +1,36 @@
-import { generate } from './generate';
-import { ProgrammaticOptions } from './types';
+import path from 'path';
+import { HandlerGenerator } from './handler/generator';
+import { MSWServerGenerator } from './msw-server/generator';
+import { TypeDefinitionGenerator } from './type-definition/generator';
+import { ApiEndpoint } from './apiEndpoint';
+import { Swagger } from './swagger';
+import { TOptions } from './types';
 
 /**
  * 프로그래밍 방식으로 MSW 핸들러를 생성합니다.
  *
  * @param options OpenAPI 문서 경로 및 출력 설정
  * @returns 생성된 핸들러 정보
  */
-async function generateMocks(options: ProgrammaticOptions) {
+async function generateMocks(options: TOptions) {
   if (!options) {
     throw new Error('Options parameter is required');
   }
-  
-  const { input, outputDir = 'src/app/mocks', ...rest } = options;
-  
-  return generate(input, {
-    input,
-    outputDir,
-    ...rest
-  });
-}
 
-export { generateMocks };
+  const swagger = await Swagger.load(options.input);
+  const apiEndpoint = new ApiEndpoint(swagger, options);
+
+  const outputFolder = options.outputDir || 'src/app/mocks';
+  const targetFolder = path.resolve(process.cwd(), outputFolder);
 
-export type { ProgrammaticOptions };
+  await new HandlerGenerator(options, apiEndpoint, swagger).generate(targetFolder);
+  await new MSWServerGenerator(options.environment).generate(targetFolder);
+  await new TypeDefinitionGenerator(apiEndpoint).generate(targetFolder);
```

Comment:
p3:
```suggestion
await Promise.all([
  new HandlerGenerator(options, apiEndpoint, swagger).generate(targetFolder),
  new MSWServerGenerator(options.environment).generate(targetFolder),
  new TypeDefinitionGenerator(apiEndpoint).generate(targetFolder),
]);
```

서로 참조관계가 없고, I/O작업이 많으니 병렬실행하면 좀 더 빠르게 작업이 가능할 것 같아요!
직접 실행해봤는데 **22.60ms**정도 단축되네요 ㅎㅎ

|              | 순차 실행 (기존 코드) | 병렬 실행 (바뀐 코드) |
|--------|--------------------|--------------------|
| 1회차    |  342.17ms              |    321.152ms            |
| 2회차    |  357.356ms             |    329.344ms          |
| 3회차    |  347.83ms             |    319.126ms         |
| 4회차    |  341.45ms             |    328.781ms            |
| 평균    |  347.20ms            |    324.60ms            |